components/MedicineSelector.jsx

```diff
@@ -0,0 +1,255 @@
+import React from 'react';
+import { View, Text, TouchableOpacity, TextInput, FlatList } from 'react-native';
+import { CheckBox } from 'react-native-elements';
+import { Ionicons } from '@expo/vector-icons';
+import Modal from 'react-native-modal';
+import { Calendar } from 'react-native-calendars';
+import MultiLineInput from '../src/app/components/Inputs/MultiLineInput';
+import DatePicker from './DatePicker';
+import { customTheme } from '../constants/themeConstants';
+const MedicineSelector = ({
+    searchInput, setSearchInput, // Props for search input state
+    selectedMedicine, setSelectedMedicine, // Props for selected medicine state
+    selectedPeriod, setSelectedPeriod, // Props for selected period state
+    selectedDays, setSelectedDays, // Props for selected days state
+    selectedType, setSelectedType, // Props for selected type state
+    mealDosage, setMealDosage, // Props for meal dosage state
+    medicines, selectMedicine, // Props for medicines list and select medicine function
+    handleAddMedicine, // Prop for adding new medicine function
+    days, meals, // Props for days and meals lists
+    modalVisible, setModalVisible, // Props for modal visibility state
+    currentMeal, setCurrentMeal, // Props for current meal state
+    handleDosageSelection, // Prop for handling dosage selection
+    dosageOptionsMap, // Prop for dosage options map
+    startDate, setStartDate, // Props for start date state
+    endDate, setEndDate, // Props for end date state
+    calendarModalVisible, setCalendarModalVisible, // Props for calendar modal visibility state
+    calendarType, setCalendarType, // Props for calendar type state
+    note, setNote // Props for note state
+}) => {
+
+    // Filtering out medicines based on search input
+    const filteredMedicines = medicines.filter(medicine =>
```

Comment:
Don't declare in the main file. Whenever the file will change its state, we will filter medicines. Do it in action creator or have useEffect filter it only once.